datafusion/sql/src/unparser/plan.rs

```diff
@@ -389,7 +389,10 @@ impl Unparser<'_> {
                 )?;
 
                 let ast_join = ast::Join {
```

Comment:
This is totally fine, but I figured I would point out another way to express this logic that you might prefer in future PRs

Instead of 

```rust
   let ast_join = ast::Join {
                    relation: match right_relation.build()? {
                        Some(relation) => relation,
                        None => return internal_err!("Failed to build right relation"),
                    },
                    join_operator: self
                        .join_operator_to_sql(join.join_type, join_constraint),
                };
```

You can do something like this with a `let .. else` to match or return

```rust
   let Some(relation) = match right_relation.build()? else {
    return internal_err!("Failed to build right relation"),
   };

   let ast_join = ast::Join {
                    relation,
                    join_operator: self
                        .join_operator_to_sql(join.join_type, join_constraint),
                };
```